src/overlays/actors/ovl_Door_Warp1/z_door_warp1.c

```diff
@@ -22,77 +45,1009 @@ const ActorInit Door_Warp1_InitVars = {
+        case WARP_GREEN:
+        case WARP_RED:
+            this->unk_194 = 0.23f;
+            this->unk_198 = 0.6f;
+            break;
+        case WARP_DESTINATION:
+            this->unk_194 = 0.0f;
+            this->unk_198 = 0.0f;
+            break;
+        case WARP_UNK_7:
+            this->scale = 100;
+            this->unk_1AE = 120;
+            this->unk_1B0 = 230;
+            this->unk_194 = 0.3f;
+            this->unk_198 = 0.3f;
+            break;
+        case WARP_BLUE_RUTO:
+        default:
+            this->unk_194 = 0.3f;
+            this->unk_198 = 0.3f;
+            break;
+    }
+
+    this->unk_19C = 0.0f;
+    this->actor.shape.yOffset = 1.0f;
+    this->warpTimer = 0;
+
+    switch (this->actor.params) {
+        case WARP_PURPLE_CRYSTAL:
+        case WARP_BLUE_RUTO:
+        case WARP_UNK_7:
+        default:
+            Lights_PointNoGlowSetInfo(&this->upperLightInfo, this->actor.world.pos.x, this->actor.world.pos.y,
+                                      this->actor.world.pos.z, 200, 255, 255, 255);
+            Lights_PointNoGlowSetInfo(&this->lowerLightInfo, this->actor.world.pos.x, this->actor.world.pos.y,
+                                      this->actor.world.pos.z, 200, 255, 255, 255);
+        case WARP_CLEAR_FLAG:
+        case WARP_SAGES:
+        case WARP_YELLOW:
+        case WARP_DESTINATION:
+        case WARP_ORANGE:
+        case WARP_GREEN:
+        case WARP_RED:
+            break;
+    }
+
+    switch (this->actor.params) {
+        case WARP_CLEAR_FLAG:
+            DoorWarp1_SetupAction(this, DoorWarp1_AwaitClearFlag);
+            break;
+        case WARP_DESTINATION:
+            if ((!(gSaveContext.entranceIndex == 0x608 ||  // sacred forest meadow
+                   gSaveContext.entranceIndex == 0x564 ||  // death mountain crater
+                   gSaveContext.entranceIndex == 0x60C ||  // lake hylia
+                   gSaveContext.entranceIndex == 0x610 ||  // desert colossus
+                   gSaveContext.entranceIndex == 0x580) && // graveyard
+                 gSaveContext.sceneSetupIndex < 4) ||
+                (GET_PLAYER(globalCtx)->actor.params & 0xF00) != 0x200) {
+                Actor_Kill(&this->actor);
+            }
+            if (Actor_WorldDistXZToActor(&player->actor, &this->actor) > 100.0f) {
+                Actor_Kill(&this->actor);
+            }
+            DoorWarp1_SetupAction(this, DoorWarp1_WarpIn);
+            break;
+        case WARP_UNK_7:
+            DoorWarp1_SetupAction(this, func_8099B020);
+            break;
+        default:
+            DoorWarp1_SetupAction(this, DoorWarp1_ActivateWarp);
+            break;
+    }
+}
+
+void DoorWarp1_SetupAdultDungeonWarp(DoorWarp1* this, GlobalContext* globalCtx) {
+    SkelAnime_Init(globalCtx, &this->skelAnime, &gWarpCrystalSkel, &gWarpCrystalAnim, NULL, NULL, 0);
+    Animation_ChangeImpl(&this->skelAnime, &gWarpCrystalAnim, 1.0f, 1.0f, 1.0f, ANIMMODE_ONCE, 40.0f, 1);
```

Comment:
Probably not supposed to call that directly.

Reply:
the dev probably should not have, but they did lol

Reply:
I mean it should be marked with ! BUG

Reply:
its not really a bug no, just using the system in a way it probably wasnt designed to

Reply:
Invest now in `@probablyNotWhatTheyIntended`, guaranteed returns